src/frontend/hooks/useGetQuestStates.ts

```diff
@@ -0,0 +1,79 @@
+import {
+  getPlaystreakQuestStatus,
+  getUserPlaystreakQueryOptions,
+  getQuestQueryOptions
+} from '@hyperplay/quests-ui'
+import { QuestLogInfo } from '@hyperplay/ui'
+import { Quest } from '@hyperplay/utils'
+import { useQueries } from '@tanstack/react-query'
+import useAuthSession from './useAuthSession'
+
+export interface UseGetQuestLogInfosProps {
+  quests?: Quest[] | null
+}
+
+export function useGetQuestStates({ quests }: UseGetQuestLogInfosProps) {
+  const { isSignedIn } = useAuthSession()
+  type getQuestQueryOptionsType = ReturnType<typeof getQuestQueryOptions>
```

Comment:
I think it's fine but it's really weird to see a type definitions inside a hook lol. Can we move it outside?